test/matching-inj/run.k

```diff
@@ -11,7 +12,10 @@ module RUN
   configuration <k> $PGM:Pgm </k>
 
   syntax AB ::= #lookup(AB, Map) [function]
-  rule #lookup(KEY:AB, KEY |-> VAL _) => VAL
+
+  rule #lookup(KEY1, KEY2 |-> VAL M) => #lookup(KEY1, M)
```

Comment:
This function should not require iteration over the "tail" of the Map. The second rule below should be equivalent.

Reply:
Hmm, so I tried to make sure it first tries to match the "normal" way, but if the "map iteration" case existed it would always try that case instead! I was seriously confused until I saw that during this test it tries to enter the map matching code twice, and it only succeeds the second time. This means I'd somehow need to make sure it tries to match "normally" multiple times before trying the "map iteration" code. Is this doable with the current matching algorithm? And I'm not even sure it makes sense for other cases.

Reply:
I thought about this a bit, and I think what we should do is sort the constraints when matching. I will make another issue for that, and it can be handled in a separate pull request that we merge before this one.

Reply:
Thanks!